meta/main.yml

```diff
@@ -13,4 +13,4 @@ galaxy_info:
   categories:
   - system
 dependencies:
-  - { role: "azavea.java" }
+  - { role: "azavea.java",  when: spark_install_java }
```

Comment:
awesome 👍   ( somehow didn't come to this idea, that `when` works here!)
This fix could also allow us to re-align our fork  (in which I just removed the whole dependency)

Reply:
@tnation14 I'm motivated to get half of this PR with the `spark_install_java`  part merged (essentially the first 3 commits), or the full changes ok too.
As I don't have committer perms here, I'ld create a new PR (first around the `spark_install_java`part),
but only if you would accept a merge 👍